rts/System/Ecs/Components/BaseComponents.h

```diff
@@ -25,7 +25,16 @@ struct Component : public BasicComponentType<T> { \
     Component(Component &&) = default; \
     Component& operator=(Component &&) = default; \
 };
-#define ALIAS_COMPONENT(Component, T) struct Component : public BasicComponentType<T> {};
+#define ALIAS_COMPONENT(Component, T) \
+struct Component : public BasicComponentType<T> { \
+    Component() = default; \
+    Component(T val){ value = val; } \
```

Comment:
any downside to moving (since `val` is a copy anyway)?
```suggestion
    Component(T val){ value = std::move(val); } \
```

Reply:
That makes sense. I can't think of reason at the moment to not do it.